helm/superset/templates/_helpers.tpl

```diff
@@ -61,6 +61,25 @@ Create chart name and version as used by the chart label.
   {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{/*
+Generate Redis URL for SSL and non-SSL protocols
+*/}}
+{{- define "superset-redis-url" -}}
+  {{- if .Values.supersetNode.connections.use_redis_ssl -}}
+    {{- if .Values.supersetNode.connections.redis_password -}}
+      rediss://:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/##DBNAME##?ssl_cert_reqs={{- .Values.supersetNode.connections.use_redis_ssl.ssl_cert_reqs | default "CERT_NONE" -}}
+    {{- else -}}
+      rediss://{env('REDIS_HOST')}:{env('REDIS_PORT')}/##DBNAME##?ssl_cert_reqs={{- .Values.supersetNode.connections.use_redis_ssl.ssl_cert_reqs | default "CERT_NONE" -}}
```

Comment:
Can we just make another couple of envvars? 

i.e. `{env('REDIS_PROTO')}` and `{env('REDIS_DB')}`

Reply:
- `REDIS_DB` is already in use for Flask Caching. And it defaults to `1` if this envvar does not exist. 
```yaml
'CACHE_REDIS_URL': f"{{- include "superset-redis-url" . | replace "##DBNAME##" "{env('REDIS_DB', 1)}" -}}",
```   
But yes, the database name is hardcoded to `0` for CeleryConfig Object. We can have and envvar for that and default it to `0` if not set.   
```python
class CeleryConfig:
  imports  = ("superset.sql_lab", )
  broker_url = f"{{- include "superset-redis-url" . | replace "##DBNAME##" "0" -}}"
  result_backend = f"{{- include "superset-redis-url" . | replace "##DBNAME##" "0" -}}"
```    
   
   
 - I don't think having envvar for Redis Protocol would make any difference. There can be  only 2 values for `REDIS_PROTO` viz. `redis` and `rediss` and they will be selected based on ssl config setting.

Reply:
Yep, I think we should add a prop in the config for `redis_db` (defaults to `0`). This should be set as an envvar which is then read in the config to setup Celery, etc. Ideally, the final result would be consistent with the existing settings.

Reply:
Sure, will do that. Just 1 question. Previously we were using 2 redis databases viz. `0` and `1`.   
   
Under CeleryConfig, the database is set to `0`
```python
class CeleryConfig:
  imports  = ("superset.sql_lab", )
  {{- if .Values.supersetNode.connections.redis_password }}
  broker_url = f"redis://:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
  result_backend = f"redis://:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
  {{- else }}
  broker_url = f"redis://{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
  result_backend = f"redis://{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
  {{- end }}
``` 
   
And Under Data Cache Config, database was set to `1`
As we don't have `REDIS_DB` envvar in current setup, the `env('REDIS_DB', 1)` will always return `1`
```python
CACHE_CONFIG = {
      'CACHE_TYPE': 'RedisCache',
      'CACHE_DEFAULT_TIMEOUT': 300,
      'CACHE_KEY_PREFIX': 'superset_',
      'CACHE_REDIS_HOST': env('REDIS_HOST'),
      'CACHE_REDIS_PORT': env('REDIS_PORT'),
      'CACHE_REDIS_PASSWORD': env('REDIS_PASSWORD'),
      'CACHE_REDIS_DB': env('REDIS_DB', 1),
}
DATA_CACHE_CONFIG = CACHE_CONFIG
```
   
   
So how do you suggest we go with that?    
1. Use 1 envvar `REDIS_DB` for both places and have it default value to `0`
2. Create 2 envvars `REDIS_DB` for DATA_CACHE_CONFIG with default `1` and `REDIS_CELERY_DB` for CeleryConfig with default `0`

Reply:
You should be able to add more env keys here: https://github.com/apache/superset/blob/master/helm/superset/templates/secret-env.yaml#L32

Reply:
Added new envvars, please check...